### ISSUE TITLE
Error when saving a model of models

### ISSUE THREAD

**Issue by tchaton** [keras-team/keras#9901]
Hello,
Here is my model topology.
When I am trying to save it but got an error
[<keras.engine.topology.InputLayer at 0x7fe85f8c8610>,
 <keras.engine.training.Model at 0x7fe846170390>,
 <nn.nn_blocks.BatchRenormalization at 0x7fe85f8910d0>,
 <keras.layers.pooling.GlobalAveragePooling2D at 0x7fe84611a450>,
 <keras.layers.core.Reshape at 0x7fe8476c93d0>,
 <keras.layers.core.Dense at 0x7fe845f99fd0>,
 <keras.layers.core.Dense at 0x7fe845ee2e50>,
 <keras.layers.merge.Multiply at 0x7fe845ef5ed0>,
 <keras.layers.core.Dropout at 0x7fe845ea8c10>,
 <keras.layers.pooling.AveragePooling2D at 0x7fe845e44c10>,
 <keras.layers.core.Dropout at 0x7fe845e93a50>,
 <keras.layers.pooling.GlobalMaxPooling2D at 0x7fe845e5b690>,
 <keras.layers.pooling.GlobalAveragePooling2D at 0x7fe845e5b9d0>,
 <keras.layers.merge.Add at 0x7fe845e02610>,
 <keras.layers.core.Dense at 0x7fe845dc3e90>,
 <nn.nn_blocks.BatchRenormalization at 0x7fe845dd4250>,
 <keras.layers.core.Dropout at 0x7fe845dd41d0>,
 <keras.layers.core.Activation at 0x7fe845d1c590>,
 <keras.layers.core.Dense at 0x7fe845d832d0>,
 <nn.nn_blocks.BatchRenormalization at 0x7fe845c00310>,
 <keras.layers.core.Dropout at 0x7fe845c65350>,
 <keras.layers.core.Activation at 0x7fe845bbdd90>,
 <keras.layers.core.Dense at 0x7fe845c2a390>]

I looked into it, but it seems that the get_config function return a keyError when called over the  <keras.engine.training.Model at 0x7fe846170390>. But I can save the weights correclty but not the architecture.
